recipes-bsp/u-boot/u-boot-iot2050.inc

```diff
@@ -36,6 +36,21 @@ U_BOOT_CONFIG_PACKAGE = "1"
 do_prepare_build_append() {
     cd ${S}
     ln -sf ../prebuild/* ${S}
+
+    case ${IOT2050_VARIANTS} in
+        *"PG1"*)
+            echo -n "Linking SBL for PG1."
+            ln -sf ../prebuild/tiboot3.bin_PG1 ${S}/tiboot3.bin
+            ;;
+        *"PG2"*)
+            echo -n "Linking SBL for PG2."
+            ln -sf ../prebuild/tiboot3.bin_PG2 ${S}/tiboot3.bin
+            ;;
+        *)
+            bbfatal "IOT2050_VARIANTS=${IOT2050_VARIANTS}, it must contain PG1 or PG2!"
+            ;;
+    esac
+
```

Comment:
See my comment above: This would move into the specific recipe file (-pg1 vs. -pg2).